pkg/sources/reconciler/awscloudwatchsource/adapter.go

```diff
@@ -55,12 +56,15 @@ type adapterConfig struct {
 var _ common.AdapterDeploymentBuilder = (*Reconciler)(nil)
 
 // BuildAdapter implements common.AdapterDeploymentBuilder.
-func (r *Reconciler) BuildAdapter(src commonv1alpha1.Reconcilable, sinkURI *apis.URL) *appsv1.Deployment {
+func (r *Reconciler) BuildAdapter(src commonv1alpha1.Reconcilable, sinkURI *apis.URL) (*appsv1.Deployment, error) {
 	typedSrc := src.(*v1alpha1.AWSCloudWatchSource)
 
 	var queries string
 	if qs := typedSrc.Spec.MetricQueries; len(qs) > 0 {
-		q, _ := json.Marshal(qs)
+		q, err := json.Marshal(qs)
+		if err != nil {
+			return nil, fmt.Errorf("serializing spec.metricQueries to JSON: %w")
+		}
```

Comment:
I used the occasion to actually return an error in the few places where we used to ignore them (3 occurrences across all TriggerMesh components).